app/state_machines/transaction_process_state_machine.rb

```diff
@@ -76,13 +82,28 @@ class TransactionProcessStateMachine
     send_new_transaction_email(transaction) if transaction.conversation.payment?
   end
 
+  # "guard_transition" is before SQL BEGIN-COMMIT block
+  # instead, "before_transition" is inside the block
+  before_transition(to: :preauthorized) do |transaction, transition|
```

Comment:
Should there be the same kind of block for transition to `:payment_intent_requires_action`? That state also is blocking availability, so it would be better if we guard when it is first entered. No?